### PR TITLE
GUVNOR-3078: [Guided Decision Table Graph] Graph is closed automatically after removing last table

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3078

Removing the last document no longer closes the editor.

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java [hunk 1]
```diff
@@ -249,166 +250,171 @@ public void onClose() {
         modeller.onClose();
     }
 
-    protected void onDecisionTableSelected( final DecisionTableSelectedEvent event ) {
-        final GuidedDecisionTableView.Presenter dtPresenter = event.getPresenter();
-        if ( dtPresenter == null ) {
+    protected void onDecisionTableSelected(final DecisionTableSelectedEvent event) {
+        final Optional<GuidedDecisionTableView.Presenter> dtPresenter = event.getPresenter();
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java]: Oh look.. here's a change...

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java [hunk 2]
```diff
@@ -249,166 +250,171 @@ public void onClose() {
...
 
         openOtherDecisionTable();
     }
 
     void openOtherDecisionTable() {
         final Set<GuidedDecisionTableView.Presenter> availableDecisionTables = modeller.getAvailableDecisionTables();
-        if ( availableDecisionTables == null || availableDecisionTables.isEmpty() ) {
-            placeManager.forceClosePlace( editorPlaceRequest );
+        if (availableDecisionTables == null || availableDecisionTables.isEmpty()) {
+            decisionTableSelectedEvent.fire(DecisionTableSelectedEvent.NONE);
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/BaseGuidedDecisionTableEditorPresenter.java]: And here's the crux of the change..

**Comment by manstis** [comment]
Hi @jomarko thanks for taking a look. 

I've moved the "radar" enabling/disabling to the ```RadarMenuBuilder```. 

I cannot however move the "validate" menu item related code to ```activateDocument(..)``` as that method is [not called](https://github.com/kiegroup/drools-wb/pull/498/files#diff-cf029670362eba75c492cbe69b21efe7R258) if there is no concrete document selected. 

Furthermore I cannot (easily) observe ```DecisionTableSelectedEvent``` events in the "validate" menu item itself (as I do for "radar" and "view" menus) as it's instantiation is burried in Uberfire's ```BasicFileMenuBuilder```. 

That said however all the enabling/disabling is in response to a ```DecisionTableSelectedEvent``` event - and the code for the "validate" menu item is in such a handler.. so I think it's OK, but open to other ideas.

**Comment by jomarko** [comment]
@manstis thanks, however, now I don't see the enabling/disabling of Validate menu nowhere. Did you removed it by accident?

**Comment by manstis** [comment]
@jomarko Nope.. it's still [there](https://github.com/kiegroup/drools-wb/pull/498/files#diff-cf029670362eba75c492cbe69b21efe7R255)!

**Pull request merged**